sql/type.go

```diff
@@ -731,7 +743,18 @@ func (t datetimeT) Convert(v interface{}) (interface{}, error) {
 	case string:
 		t, err := time.Parse(DatetimeLayout, value)
 		if err != nil {
-			return nil, ErrConvertingToTime.Wrap(err, v)
+			failed := true
+			for _, fmt := range DatetimeLayouts {
+				if t2, err2 := time.Parse(fmt, value); err2 == nil {
+					t = t2
+					failed = false
+					break
+				}
+			}
+
+			if failed {
+				return nil, ErrConvertingToTime.Wrap(err, v)
```

Comment:
This wrapped error is no longer applicable (it will give the false impression that the last format attempted is the only correct one)